runtime/vm/resolvesupport.cpp

```diff
@@ -496,7 +498,71 @@ resolveStaticMethodRefInto(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cp
 	if (isResolvedClassAnInterface) {
 		lookupOptions |= J9_LOOK_INTERFACE;
 	}
-	method = (J9Method *)javaLookupMethod(vmStruct, resolvedClass, J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef), cpClass, lookupOptions);
+
+	#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	if (resolvedClass == J9VMJAVALANGINVOKEMETHODHANDLE(vmStruct->javaVM)) {
+		J9UTF8 *nameUTF = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+		/**
+		 * Check for MH intrinsic methods
+		 *
+		 * Modify the signature to avoid signature mismatch due to varargs
+		 * These methods have special INL send targets
+		 */
+		U_8* initialMethodName = J9UTF8_DATA(nameUTF);
+		U_16 initialMethodNameLength = J9UTF8_LENGTH(nameUTF);
+		BOOLEAN isMethodHandle = FALSE;
+
+		switch (initialMethodNameLength) {
+		case 11:
+			if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "invokeBasic")) {
+				isMethodHandle = TRUE;
+			}
+			break;
+		case 12:
+			if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToStatic")) {
+				isMethodHandle = TRUE;
+			}
+			break;
+		case 13:
+			if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToSpecial")
+			||  J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToVirtual")) {
+				isMethodHandle = TRUE;
+			}
+			break;
+		case 15:
+			if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToInterface")) {
+				isMethodHandle = TRUE;
+			}
+			break;
+		}
+
+		if (isMethodHandle) {
+			/* Stack allocate a byte array for MethodHandle method name and signature. The array size is:
+			*  - J9ROMNameAndSignature
+			*  - Modified method name
+			*      - U_16 for J9UTF8 length
+			*      - 16 bytes for the original method name ("linkToInterface" is the longest)
+			*  - J9UTF8 for empty signature
+			*/
+			U_8 nameAndNAS[sizeof(J9ROMNameAndSignature) + (sizeof(U_16) + 16) + sizeof(J9UTF8)];
```

Comment:
This can't be stack allocated here and assigned to `nameAndSig` as `nameAndSig` has a longer live range:
```
nameAndSig = (J9ROMNameAndSignature *)nameAndNAS;
```

This will corrupt memory as the stack buffer is only valid till the end of the scope it's defined in